bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -150,6 +162,8 @@ def run_inner_plan(scope, plan_name, targets, args = {})
           raise e
         end
       ensure
+        # Pop the plan invocation ID off of the plan_id stack for the Future.
+        current_future&.plan_stack&.shift
```

Comment:
This doesn't really matter since the effect is the same, but why not use `#pop` to pop elements off the stack? :P Took a second to realize the `plan_stack` is backwards from what I expected when examining it.

Reply:
Honestly only did it this way because I spent *so long* looking at the [Pops::PuppetStack](https://github.com/puppetlabs/puppet/blob/main/lib/puppet/pops/puppet_stack.rb) class + stack while trying to make the scope work for the Plan ID. I can change it if it's more readable, I think it was just on the brain.

Reply:
Nah, I don't think it really matters. 🤷